zhaquirks/tuya/mcu/__init__.py

```diff
@@ -548,6 +563,154 @@ class MoesSwitchManufCluster(TuyaOnOffManufCluster):
+                # make sure this is sent as a enum, regardless that tuya_cover_command probably
+                # contains ints.
+                None,
+                t.enum8
+            ),
+            2: DPToAttributeMapping(
+                TuyaNewWindowCoverControl.ep_attribute,
+                ATTR_COVER_LIFTPERCENT_CONTROL_NAME,
+                self.convert_lift_percent,
+                self.convert_lift_percent
+            ),
+            3: DPToAttributeMapping(
+                TuyaNewWindowCoverControl.ep_attribute,
+                ATTR_COVER_LIFTPERCENT_NAME,
+                self.convert_lift_percent,
+                self.convert_lift_percent
+            ),
+            13: DPToAttributeMapping(
+                PowerConfiguration.ep_attribute,
+                PowerConfiguration.AttributeDefs.battery_percentage_remaining.name
+                # Tuya report real percent, zigbee expects value*2, but
+                # TuyaPowerConfigurationCluster will convert it
+            ),
+        }
+
+    data_point_handlers = {
+        1: "_dp_2_attr_update",
+        2: "_dp_2_attr_update",
+        3: "_dp_2_attr_update",
+        # Define DPs 5 & 7 to avoid warning logs, but I'm not sure what they are so just ignore
+        # updates from them
+        5: "ignore_update",
+        7: "ignore_update",
+        13: "_dp_2_attr_update",
+    }
+
+    def convert_lift_percent(self, input_value: int):
+        """Convert/invert lift percent when needed.
+
+        HA shows % open. The zigbee cluster value is called 'lift_percent' but seems to need to
+        be % closed. This logic follows the convention of other Tuya covers, inverting the value
+        by default and reversing that invert if tuya_cover_inverted_by_default or the cluster
+        invert attribute is set. (This seems strange to me, but it's better to be consistent.)
+        """
+
+        invert_attr = self.endpoint.window_covering._attr_cache.get(ATTR_COVER_INVERTED) == 1
+        invert = (
+            not invert_attr
+            if self.endpoint.device.tuya_cover_inverted_by_default
+            else invert_attr
+        )
+        return input_value if invert else 100 - input_value
+
+    def ignore_update(self, _datapoint: TuyaDatapointData) -> None:
+        """Process (and ignore) some data point updates."""
+        return None
+
+
+class TuyaNewWindowCoverControl(TuyaLocalCluster, WindowCovering):
+    """Tuya Window Cover Cluster, based on new TuyaNewManufClusterForWindowCover.
+
+    Derive from TuyaLocalCluster to disable attribute writes, in a way that's compatible with
+    TuyaNewManufClusterForWindowCover (not TuyaManufacturerWindowCover.)
+    """
+
+    attributes = WindowCovering.attributes.copy()
+    # cover direction and lift percent control attributes are not very useful to HA, but needed to
+    # allow TuyaMCUCluster to map to the data point.
+    attributes.update({ATTR_COVER_DIRECTION: (ATTR_COVER_DIRECTION_NAME, t.enum8)})
```

Comment:
I'm surprised I need a new attribute to track motor status (i.e. opening, closing, stopped) but I couldn't find anything in the standard zigbee WindowCovering.